### DISCUSSION TITLE
Segmenting validation with different modes, such as having both onBlur and onSubmit validation separately?

### DISCUSSION THREAD

**Discussion by karrui** [react-hook-form/react-hook-form#9066] [Ideas]
There is currently field level validation, but sometimes one might want certain errors to only show up on submit.
Is there a possibility of segmenting validation rules by a mode, similar to the mode in `useForm`? 

One use case that we would use this for could be something like a verifiable field (with multiple inputs), where we do not want to show an error (since user is still performing a group action) until the form is attempting to submit.

For example, the below input should be a field level validation and trigger normally (`onChange`, `onBlur`, `onTouch`, whatever is set on `useForm`)
![Screenshot 2022-09-20 at 11 34 27 AM](https://user-images.githubusercontent.com/22133008/191161698-197e38e3-8d19-4849-b5b0-c6761fac03ab.png)

However, the below input should not error when user is pending an action that they are in the midst of for better UX:
![Screenshot 2022-09-20 at 11 36 40 AM](https://user-images.githubusercontent.com/22133008/191162061-307c152e-a63f-4803-98d5-082c3d33f788.png)

It would help if there is a way to only trigger certain validations specifically on a different mode, such as onSubmit.

**Comment by kmfb** [comment]
Your problem just like [issue](https://github.com/react-hook-form/react-hook-form/issues/9064#issuecomment-1250853355) which i created, and the [antd](https://ant.design/components/form/#Rule) support the field validation because they use rule config as a array not a object in RHF, with the validateTrigger, you can validate by field in different mode, not just the whole form validation mode.
So there is may not a method to validate field by different mode until the maintainer supports the feature in this project.

**Comment by kmfb** [comment]
> There is currently field level validation, but sometimes one might want certain errors to only show up on submit. Is there a possibility of segmenting validation rules by a mode, similar to the mode in `useForm`?
> 
> One use case that we would use this for could be something like a verifiable field (with multiple inputs), where we do not want to show an error (since user is still performing a group action) until the form is attempting to submit.
> 
> For example, the below input should be a field level validation and trigger normally (`onChange`, `onBlur`, `onTouch`, whatever is set on `useForm`) ![Screenshot 2022-09-20 at 11 34 27 AM](https://user-images.githubusercontent.com/22133008/191161698-197e38e3-8d19-4849-b5b0-c6761fac03ab.png)
> 
> However, the below input should not error when user is pending an action that they are in the midst of for better UX: ![Screenshot 2022-09-20 at 11 36 40 AM](https://user-images.githubusercontent.com/22133008/191162061-307c152e-a63f-4803-98d5-082c3d33f788.png)
> 
> It would help if there is a way to only trigger certain validations specifically on a different mode, such as onSubmit.

But in your case, I think you can do this way, don't set the rule for the sms code validation, in handleSubmit method, which you can use setError by the input value conditions.
